### ISSUE TITLE
`vendor/autoload.php` is unnecessarily changed every time you run composer update

### ISSUE THREAD

**Issue by fulldecent** [composer/composer#3701]
The autoloader name is set randomly each time you run `composer update`. This produces a change like the following which serves no benefit. I am requesting that this file to not change each time you run `composer update`.

````
diff --git a/vendor/autoload.php b/vendor/autoload.php
index 58d2646..4ad48cb 100644
--- a/vendor/autoload.php
+++ b/vendor/autoload.php
@@ -4,4 +4,4 @@
 
 require_once __DIR__ . '/composer' . '/autoload_real.php';
 
-return ComposerAutoloaderInit20ab832d70b7ac430b7d63746516d8ba::getLoader();
+return ComposerAutoloaderInit6486675c13c285b2f89c612eb53e9d2f::getLoader();
diff --git a/vendor/composer/autoload_real.php b/vendor/composer/autoload_real.php
index b2b4090..3a808ef 100644
--- a/vendor/composer/autoload_real.php
+++ b/vendor/composer/autoload_real.php
@@ -2,7 +2,7 @@
 
 // autoload_real.php @generated by Composer
 
-class ComposerAutoloaderInit20ab832d70b7ac430b7d63746516d8ba
+class ComposerAutoloaderInit6486675c13c285b2f89c612eb53e9d2f
 {
     private static $loader;
 
@@ -19,9 +19,9 @@ class ComposerAutoloaderInit20ab832d70b7ac430b7d63746516d8ba
             return self::$loader;
         }
 
-        spl_autoload_register(array('ComposerAutoloaderInit20ab832d70b7ac430b7d63746516d8ba', 'loadClassLoader'), true, true);
+        spl_autoload_register(array('ComposerAutoloaderInit6486675c13c285b2f89c612eb53e9d2f', 'loadClassLoader'), true, true);
         self::$loader = $loader = new \Composer\Autoload\ClassLoader();
-        spl_autoload_unregister(array('ComposerAutoloaderInit20ab832d70b7ac430b7d63746516d8ba', 'loadClassLoader'));
+        spl_autoload_unregister(array('ComposerAutoloaderInit6486675c13c285b2f89c612eb53e9d2f', 'loadClassLoader'));
 
         $map = require __DIR__ . '/autoload_namespaces.php';
         foreach ($map as $namespace => $path) {
@@ -44,7 +44,7 @@ class ComposerAutoloaderInit20ab832d70b7ac430b7d63746516d8ba
     }
 }
 
-function composerRequire20ab832d70b7ac430b7d63746516d8ba($file)
+function composerRequire6486675c13c285b2f89c612eb53e9d2f($file)
 {
     require $file;
 }
````

Why do I want this?
=================

Because I include `/vendor` files in my repository and this behavior pollutes my changelog. Please, please, **please**, do not tell me that I should change my workflow to exclude `/vendor` from my repository. I know, we have different workflows. Please note that the composer page on "Should I commit the dependencies in my vendor directory?" includes a section "If you really feel like you must do this...", which indicates I'm not the only person doing this. Also the page http://guides.cocoapods.org/using/using-cocoapods.html from a similar product also discusses use cases where people may or may not want to have dependencies in their repositories.

**Comment by GromNaN** [comment]
Does it fix you issue if you configure the [`autoloader-suffix`](https://getcomposer.org/doc/04-schema.md#config) ?

> `autoloader-suffix`: Defaults to null. String to be used as a suffix for the generated Composer autoloader. When null a random one will be generated.

For reference, this change was introduced by https://github.com/composer/composer/pull/963.

**Comment by slbmeh** [comment]
TL;DR; for the referenced issue, the random string suffix is to break opcode caching in PHP.

**Comment by fulldecent** [comment]
Thanks, this fix works great and implements good default behavior.
